File: rust/src/models/payment_method_info.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::payment_method_type_enum::PaymentMethodTypeEnum;

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct PaymentMethodInfo {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub account_number_hint: Option<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub card_brand: Option<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub card_last4: Option<String>,

    pub payment_method_type: PaymentMethodTypeEnum,
}

impl PaymentMethodInfo {
    pub fn new(payment_method_type: PaymentMethodTypeEnum) -> Self {
        Self {
            account_number_hint: None,
            card_brand: None,
            card_last4: None,
            payment_method_type,
        }
    }
}

File: typescript/src/models/paymentMethodInfo.ts
// this file is @generated
import {
  type PaymentMethodTypeEnum,
  PaymentMethodTypeEnumSerializer,
} from "./paymentMethodTypeEnum";

export interface PaymentMethodInfo {
  accountNumberHint?: string | null;

  cardBrand?: string | null;

  cardLast4?: string | null;

  paymentMethodType: PaymentMethodTypeEnum;
}

export const PaymentMethodInfoSerializer = {
  _fromJsonObject(object: any): PaymentMethodInfo {
    return {
      accountNumberHint: object["account_number_hint"],
      cardBrand: object["card_brand"],
      cardLast4: object["card_last4"],
      paymentMethodType: PaymentMethodTypeEnumSerializer._fromJsonObject(
        object["payment_method_type"]
      ),
    };
  },

  _toJsonObject(self: PaymentMethodInfo): any {
    return {
      account_number_hint: self.accountNumberHint,
      card_brand: self.cardBrand,
      card_last4: self.cardLast4,
      payment_method_type: PaymentMethodTypeEnumSerializer._toJsonObject(
        self.paymentMethodType
      ),
    };
  },
};
